modin/pandas/test/dataframe/test_protocol.py

```diff
@@ -0,0 +1,276 @@
+
+    assert col.size == 5
+    assert col.null_count == 1
+    assert col.dtype[0] == DTypeKind.STRING
+    if StorageFormat.get() != "Omnisci":
+        assert col.describe_null == (4, 0)
+
+    df2 = df.__dataframe__()
+    assert df2.A.tolist() == df.A.tolist()
+    assert df2.get_column_by_name("A").null_count == 1
+    if StorageFormat.get() != "Omnisci":
+        assert df2.get_column_by_name("A").describe_null == (4, 0)
+    assert df2.get_column_by_name("A").dtype[0] == DTypeKind.STRING
+
+    df_sliced = df[1:]
+    col = df_sliced.__dataframe__().get_column_by_name("A")
+    assert col.size == 4
+    assert col.null_count == 1
+    assert col.dtype[0] == DTypeKind.STRING
+    if StorageFormat.get() != "Omnisci":
+        assert col.describe_null == (4, 0)
+
+    df2 = df_sliced.__dataframe__()
+    assert df2.A.tolist() == df_sliced.A.tolist()
+    assert df2.get_column_by_name("A").null_count == 1
+    if StorageFormat.get() != "Omnisci":
+        assert df2.get_column_by_name("A").describe_null == (4, 0)
+    assert df2.get_column_by_name("A").dtype[0] == DTypeKind.STRING
+
+
+def test_object():
+    df = pd.DataFrame({"x": np.array([None, True, False])})
+    col = df.__dataframe__().get_column_by_name("x")
+
+    assert col.size == 3
+
+    with pytest.raises(NotImplementedError):
+        assert col.dtype
+    with pytest.raises(NotImplementedError):
+        assert col.describe_null
+
+
+def test_DataFrame():
+    df = pd.DataFrame(
+        {
+            "x": np.array([True, True, False]),
+            "y": np.array([1, 2, 0]),
+            "z": np.array([9.2, 10.5, 11.8]),
+        }
+    )
+
+    df2 = df.__dataframe__()
+
+    assert df2._allow_copy is True
+    assert df2.num_columns() == 3
+    assert df2.num_rows() == 3
+    assert df2.num_chunks() == 1
+
+    assert list(df2.column_names()) == ["x", "y", "z"]
+
+    assert (
+        df2.select_columns((0, 2))._df[:, 0].tolist()
+        == df2.select_columns_by_name(("x", "z"))._df[:, 0].tolist()
+    )
+    assert (
+        df2.select_columns((0, 2))._df[:, 1].tolist()
+        == df2.select_columns_by_name(("x", "z"))._df[:, 1].tolist()
+    )
+
+
+def test_chunks():
+    df = pd.DataFrame({"x": np.arange(10)})
+    df2 = df.__dataframe__()
+    chunk_iter = iter(df2.get_chunks(3))
+    chunk = next(chunk_iter)
+    assert chunk.num_rows() == 4
+    chunk = next(chunk_iter)
+    assert chunk.num_rows() == 4
+    chunk = next(chunk_iter)
+    assert chunk.num_rows() == 2
+    with pytest.raises(StopIteration):
+        chunk = next(chunk_iter)
+
+
+def test_categorical_dtype():
+    df = pd.DataFrame({"A": [1, 2, 5, 1]})
+    df["A"].astype("category")
+    col = df.__dataframe__().get_column_by_name("A")
+    assert col.dtype[0] == DTypeKind.CATEGORICAL
+    assert col.describe_categorical == (False, True, {0: 1, 1: 2, 2: 5})
+
+
+def test_NA_categorical_dtype():
+    df = pd.DataFrame({"A": [1, 2, 5, 1]})
+    df["B"] = df["A"].astype("category")
```

Comment:
what's the difference between these two tests (`test_categorical_dtype` and `test_NA_categorical_dtype`)? Why can't we check for `null_count == 0` and `num_chunks == 1` in the first test?

Reply:
@dchigarev I added this checks in ``test_categorical_dtype`` and removed ``test_NA_categorical_dtype``.